test/topicFilters.spec.ts

```diff
@@ -0,0 +1,296 @@
+    await subscriptionClient.addRule(
+      "Priority_1",
+      "(priority = 1 OR priority = 3) AND (sys.label LIKE '%String1')",
+      "SET sys.body = 'MessageX'"
+    );
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "Priority_1");
+  });
+
+  it("Add Rule with SQL filter and action", async function(): Promise<void> {
+    await subscriptionClient.addRule(
+      "Priority_1",
+      "(priority = 1 OR priority = 3) AND (sys.label LIKE '%String1')",
+      "SET sys.body = 'MessageX'"
+    );
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "Priority_1");
+  });
+
+  it("Add Rule with Correlation filter", async function(): Promise<void> {
+    await subscriptionClient.addRule("Correlationfilter", {
+      label: "red",
+      correlationId: "high"
+    });
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "Correlationfilter");
+  });
+
+  it("Add Default rule on a subscription that its default rule removed", async function(): Promise<
+    void
+  > {
+    await subscriptionClient.addRule("$Default", {
+      label: "red",
+      correlationId: "high"
+    });
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "$Default");
+  });
+
+  it("Adding a rule with a name which matches with existing rule", async function(): Promise<void> {
+    await subscriptionClient.addRule("Priority_1", "priority = 1");
+    let errorWasThrown = false;
+    try {
+      await subscriptionClient.addRule("Priority_1", "priority = 2");
+    } catch (error) {
+      errorWasThrown = true;
+      should.equal(!error.message.search(" already exists."), false);
+      console.log(error.message);
+    }
+    should.equal(errorWasThrown, true);
+  });
+
+  it("Adding a rule with no name", async function(): Promise<void> {
+    let errorWasThrown = false;
+    try {
+      await subscriptionClient.addRule("", "priority = 2");
+    } catch (error) {
+      errorWasThrown = true;
+      should.equal(!error.message.search("name is missing"), false);
+      console.log(error.message);
+    }
+    should.equal(errorWasThrown, true);
+  });
+
+  it("Adding a rule with no filter", async function(): Promise<void> {
+    let errorWasThrown = false;
+    try {
+      await subscriptionClient.addRule("Priority_1", "");
+    } catch (error) {
+      errorWasThrown = true;
+      should.equal(!error.message.search("Filter is missing"), false);
+      console.log(error.message);
+    }
+    should.equal(errorWasThrown, true);
+  });
+
+  it("Adding a rule with a Boolean filter whose input is not Boolean", async function(): Promise<
+    void
+  > {
+    let errorWasThrown = false;
+    try {
+      await subscriptionClient.addRule("Priority_1", "priority = 1");
+      await subscriptionClient.addRule("Priority_2", "1");
+    } catch (error) {
+      errorWasThrown = true;
```

Comment:
What error is thrown here?

Reply:
For the test `"Adding a rule with a Boolean filter whose input is not Boolean"`

> { InternalServerError: The service was unable to process the request; please retry the operation. For more information on exception types and proper exception handling, please refer to http://go.microsoft.com/fwlink/?LinkId=761101, SystemTracker:NoSystemTracker, Timestamp:2019-01-15T19:03:45

Checked many times, this is the error I see. 
Are we expecting something like "Cannot accept this type of expression" ?

Reply:
Can you check which of the 2 is failing? Is it `await subscriptionClient.addRule("Priority_1", "priority = 1");` or `await subscriptionClient.addRule("Priority_2", "1")`;

I believe the first one shouldnt fail.

Reply:
The error is caused by the second one for sure.